src/Game/Scripting/Commands.cs

```diff
@@ -24,845 +24,1062 @@
+    public class Command
+    {
+        // All commands share access to the execution queues
+        public static Dictionary<Attributes, Queue<CommandExecution>> Queues = new Dictionary<Attributes, Queue<CommandExecution>>();
 
-        //        if (item.ItemID == find)
-        //        {
-        //            PlayerData.DoubleClick(item);
-        //            return true;
-        //        }
-        //        else if (item.Contains != null && item.Contains.Count > 0)
-        //        {
-        //            if (UseItem(item, find))
-        //                return true;
-        //        }
-        //    }
+        // Delegate to allow customization of how execution (both action logic and wait logic) are performed
+        public delegate bool Handler(CommandExecution execution);
 
-        //    return false;
-        //}
+        // Basic usage syntax
+        public string Usage { get; }
 
-        private static bool SetAbility(string command, Argument[] args, bool quiet, bool force)
-        {
-            var ability = args[0].AsString();
+        // Keyword in the script syntax
+        public string Keyword { get; }
 
-            if (args.Length < 1 || !abilities.Contains(ability))
-            {
-                throw new RunTimeError(null, "Usage: setability ('primary'/'secondary'/'stun'/'disarm') ['on'/'off']");
-            }
+        // Name of parameters used by this command
+        public string[] ParamNames { get; }
 
-            if (args.Length == 2 && args[1].AsString() == "on" || args.Length == 1)
-            {
-                switch (ability)
-                {
-                    case "primary":
-                        GameActions.UsePrimaryAbility();
-                        break;
-                    case "secondary":
-                        GameActions.UseSecondaryAbility();
-                        break;
-                    case "stun":
-                        GameActions.RequestStun();
-                        break;
-                    case "disarm":
-                        GameActions.RequestDisarm();
-                        break;
-                    default:
-                        break;
-                }
-            }
-            else if (args.Length == 2 && args[1].AsString() == "off")
-            {
-                GameActions.ClearAbility();
-            }
+        // Handler to perform command logic (action of the command)
+        public Handler ExecutionLogic { get; }
 
-            return true;
-        }
+        // Handler to perform wait logic (deciding if command can execute)
+        public Handler WaitLogic { get; }
 
-        private static bool Attack(string command, Argument[] args, bool quiet, bool force)
+        // Defines what to command related to when it comes to game logic
+        // Also used to organize execution queues
+        [Flags]
+        public enum Attributes
         {
```

Comment:
This may be overengineering, as honestly, I am not sure about the details/limitation with UO Steam.
But when I tried playing with how actions block or not others, I noticed many Actions are not blocking. For example, my current Razor Macro for training carpentry is never blocking my capacity to execute commands to find another target or even using many of my Hotkeys for spells, potions and more.

In a very high-level, that gave me the notion of 3 queues: actions to objects (including to player, such as using skills), movement, and targeting. But, because of me not being sure (basically I still very noob), and because I wanted this little engine to be ready for whatever command we add to it, I ended up creating those Flags.

Of course, I will be happy to remove it if you think is overcomplicating things 😊